content/chainguard/migration/dockerfile-conversion.md

```diff
@@ -0,0 +1,233 @@
+
+```
+dfc --help
+```
+
+You will receive output with basic usage instructions.
+
+```
+Usage:
+  dfc [flags]
+
+Examples:
+dfc <path_to_dockerfile>
+
+Flags:
+  -h, --help         help for dfc
+  -i, --in-place     modified the Dockerfile in place (vs. stdout), saving original in a .bak file
+      --json         print dockerfile as json (pre-conversion)
+      --org string   the root repo namespace at cgr.dev/<org> (default "ORGANIZATION")
+```
+
+## Basic Usage
+
+Unless specified, dfc will not make any direct changes to your Dockerfile, writing the results to the default output stream. Run the following to convert a Dockerfile and save the output to a new file:
+
+```
+dfc ./Dockerfile > ./Dockerfile.converted
+```
+
+You can also pipe the Dockerfile’s contents from stdin:
+
+```
+cat ./Dockerfile | dfc -
+```
+
+The following CLI flags are available:
+
+* `--org=<org>` \- the registry namespace, i.e. `cgr.dev/<org>` (default placeholder: `ORGANIZATION`)
+* `--json` \- serialize Dockerfile as JSON
+* `--in-place / -i` \- modify the Dockerfile in place vs. printing to stdout, saving original in a .bak file
+
+## Examples
+
+This section has a few practical examples you can use as reference.
+
+### Setting the ORG
+
+By default, `dfc` uses `ORGANIZATION` as a placeholder for the image registry address. You can provide the `--org` parameter to specify the organization that you’re a member of. To use free tier images, use `chainguard` as the organization.
+
+Consider the following Dockerfile for a CLI PHP application:
+
+```
+FROM php:8.2-cli
+
+RUN apt-get update && apt-get install -y \
+    git \
+    curl \
+    libxml2-dev \
+    zip \
+    unzip
+
+# Install Composer and set up application
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+RUN mkdir /application
+COPY . /application/
+RUN cd /application && composer install
+
+ENTRYPOINT [ "php", "/application/minicli" ]
+```
+
+The following command will convert this Dockerfile to use Chainguard Images, using the `chainguard` org for the free tier images. The output will be redirected to a new file called `Dockerfile.new`:
+
+```
+dfc Dockerfile > Dockerfile.new --org chainguard
+```
+
+The modified file will now use `cgr.dev/chainguard/php:latest-dev` as base image:
+
+```
+FROM cgr.dev/chainguard/php:latest-dev
+USER root
+
+RUN apk add -U curl git libxml2-dev unzip zip
+
+# Install Composer and set up application
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+RUN mkdir /application
+COPY . /application/
+RUN cd /application && composer install
+
+ENTRYPOINT [ "php", "/application/minicli" ]
+```
+
+### Inline Usage
+
+With inline usage, you can convert single instructions or entire Dockerfiles. For example, to convert a single `FROM` line, you can run:
+
+```
+echo "FROM node" | dfc --org chainguard.edu -
+```
+
+This will give you the following result:
+
+```
+FROM cgr.dev/chainguard.edu/node:latest-dev
+USER root
```

Comment:
This line isn't showing up for me, though I might not have the latest version of dfc installed

Reply:
good catch! it's from the latest update. I will remove the USER line